### PR TITLE
refactor(Listbox): Keyboard navigation

### PR DESCRIPTION
**Pull request by johanlahti**
## Motivation

Enable keyboard navigation, permitting the following features:
- **Space**: Select values with space
- **Arrow buttons**: Move to previous/next field
- **Arrow buttons + Shift**: Select a range
- **Enter**: Confirm selections
- **Escape**: Remove focus from `document.activeElement` and cancel selections

Already supported:
- **Tab**: Supported "natively" using tabIndex where needed
- Focus search field on toggle

### CODE REVIEW

**Comment by Caele** [comment]
LGTM, make sure to update from master as I updated the linting rules.

**Comment by johanlahti** [comment]
I have merged in latest master and fixed some breaking code & unit tests in response to this.

**Comment by johanlahti** [comment]
I have merged in latest master and fixed some breaking code & unit tests in response to this.

File: apis/nucleus/src/components/listbox/ListBoxRowColumn.jsx [hunk 1]
```diff
@@ -153,10 +157,13 @@ export default function RowColumn({ index, style, data, column = false }) {
     checkboxes = false,
     dense = false,
     frequencyMode = 'N',
+    actions,
     frequencyMax = '',
     histogram = false,
   } = data;
 
+  const handleKeyDownCallback = useCallback(getKeyboardNavigation(actions), [actions]);
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/ListBoxRowColumn.jsx:165]: I think the useCallback here will have very limited effect as the the parent component send in a new actions object every times it re-render. (It still could help for the case when only this and not the parent component is updated)

File: apis/nucleus/src/components/listbox/listbox-keyboard-navigation.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        break;
+      case KEYS.ENTER:
+        confirm();
+        break;
+      case KEYS.ESCAPE:
+        cancel();
+        if (document.activeElement) {
+          document.activeElement.blur();
+        }
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/listbox-keyboard-navigation.js:53]: In most real use cases I expect that you want to move the focus somewhere else (outside the listbox) and not just remove it. (but the listbox itself will not have enought information)

**Review by T-Wizard — APPROVED** [review]
With this pressing enter in a popover listbox confirm the selection but do not close the listbox.
I think that is a bit confusion and I woud expect it to close the popover.

File: apis/nucleus/src/components/listbox/ListBoxRowColumn.jsx (hunk 1 above)
Comment (johanlahti) [reply apis/nucleus/src/components/listbox/ListBoxRowColumn.jsx:165]: Yeah agreed. I wrap this mostly as a best practice/habit than intentional optimisation.

**Comment by Caele** [comment]
@johanlahti I now noticed you are not using the accessibility settings at all? Not sure how it works in the listbox but you need to make use of the keyboard hook I think. 
It has a lot of functions to be used for recieving focus and bluring, currently only used in the table.

**Comment by Caele** [comment]
> @Caele I tried using it in sense-client but since this is not a Supernova I can't call hooks from there. (I'm using the `field` method to create the Listbox). Is the `useKeyboard` hook intended for use from within Nebula as well? I don't see any usage yet inside Nebula.js, maybe because it's new…

It was develop for usage in the table, so no other places yet. You should be able to get hold of the object it uses to track things at least, but I'd have to test a bit to know the details.

What is important is that the listbox follows the same principles as other supernovas do, when turning accessibility on/off and how blur/focus is handled.

File: apis/nucleus/src/components/listbox/listbox-keyboard-navigation.js (hunk 2 above)
Comment (johanlahti) [reply apis/nucleus/src/components/listbox/listbox-keyboard-navigation.js:55]: Yeah, maybe this is where the `useKeyboard` can help. I think @Caele might have a look next week and see how we can use the keyboard. I am far from 100% sure about the accessibility best practices here so may be more holes to fill. We can improve iteratively.

**Pull request merged**